packages/flutter/test/material/navigation_bar_test.dart

```diff
@@ -0,0 +1,429 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  setUp(() async {
+    WidgetsBinding.instance!.focusManager.highlightStrategy =
+        FocusHighlightStrategy.alwaysTraditional;
+  });
+
+  testWidgets('Navigation bar updates destinations when tapped', (WidgetTester tester) async {
+    int mutatedIndex = 0;
+    final Widget widget = _buildWidget(
+      UniqueKey(),
+      NavigationBar(
+        destinations: const <Widget>[
+          NavigationDestination(
+            icon: Icon(Icons.ac_unit),
+            label: 'AC',
+          ),
+          NavigationDestination(
+            icon: Icon(Icons.access_alarm),
+            label: 'Alarm',
+          ),
+        ],
+        onDestinationSelected: (int i) {
+          mutatedIndex = i;
+        },
+      ),
+    );
+
+    await tester.pumpWidget(widget);
+    await tester.tap(find.text('Alarm'));
+
+    expect(mutatedIndex, 1);
+  });
+
+  testWidgets('NavigationBar content test', (WidgetTester tester) async {
```

Comment:
This test could be combined with the previous one. They're essentially a "smoke test".